File: workbench/node-native-probe/panic-probe.mjs
import assert from 'node:assert/strict';

const { nativePanicProbe } = await import('./binding.js');

await assert.rejects(
  () => nativePanicProbe(),
  (error) =>
    error instanceof Error &&
    error.message.includes('WORKFLOW_NATIVE_ERROR:') &&
    error.message.includes('intentional native task panic probe')
);
process.stdout.write('native panic became a Promise rejection\n');

File: workbench/python-native-probe/panic_probe.py
from __future__ import annotations

import asyncio

from workflow_python_native_probe import WorkflowNativeError, native_panic_probe


async def main() -> None:
    try:
        await native_panic_probe()
    except WorkflowNativeError as error:
        assert error.kind == "panic"
        assert "intentional native task panic probe" in str(error)
    else:
        raise AssertionError("native panic should become a Python exception")
    print("native panic became an awaitable exception")


asyncio.run(main())
